Client/Application/AppDelegate.swift

```diff
@@ -865,6 +871,15 @@ extension AppDelegate {
         // For now, we should just re-process the message handling.
         let handler = FxAPushMessageHandler(with: profile)
         handler.handle(userInfo: userInfo).upon { res in
+            if let message = res.successValue {
+                switch message {
+                case .thisDeviceDisconnected:
+                    FxALoginHelper.sharedInstance.applicationDidDisconnect(application)
```

Comment:
When testing this against ios-push2, got this error while the settings page was up and I disconnected from another client.

```
2017-08-15 13:40:42.881 [Info] [PushClient.swift:156] send(request:) > DELETE https://updates.push.services.mozilla.com/v1/apns/fennec/registration/05412ef0eddb4438aac45cdc392cd035
2017-08-15 13:40:42.920280-0400 Client[2582:697319] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   CoreFoundation                      0x0000000182ceeff8 <redacted> + 148
	1   libobjc.A.dylib                     0x0000000181750538 objc_exception_throw + 56
	2   CoreFoundation                      0x0000000182ceef28 <redacted> + 0
	3   Foundation                          0x00000001838e1378 <redacted> + 128
	4   Foundation                          0x0000000183729a38 <redacted> + 36
	5   UIKit                               0x0000000188e1c3b0 <redacted> + 816
	6   UIKit                               0x0000000189115ed0 <redacted> + 172
	7   UIKit                               0x0000000188e1b434 <redacted> + 512
	8   UIKit                               0x0000000188ff791c <redacted> + 220
	9   UIKit                               0x0000000188f993c8 <redacted> + 512
	10  UIKit                               0x0000000189154eec <redacted> + 116
	11  UIKit                               0x000000018915b7fc <redacted> + 188
	12  CoreFoundation                      0x0000000182bdde34 <redacted> + 132
	13  CoreFoundation                      0x0000000182bddd30 <redacted> + 312
	14  UIKit                               0x0000000188eff7ec <redacted> + 1432
	15  Client                              0x00000001002f71a8 _TFC6Client27SettingsTableViewController26SELfirefoxAccountDidChangefT_T_ + 176
	16  Client                              0x00000001002f71f4 _TToFC6Client27SettingsTableViewController26SELfirefoxAccountDidChangefT_T_ + 40
```